yt/frontends/enzo_e/fields.py

```diff
@@ -128,33 +128,40 @@ def setup_particle_fields(self, ptype, ftype="gas", num_neighbors=64):
         self.setup_particle_mass_field(ptype)
 
     def setup_particle_mass_field(self, ptype):
-        name = "particle_mass"
+        fname = "particle_mass"
         if ptype in self.ds.particle_unions:
-            add_union_field(self, ptype, name, "code_mass")
+            add_union_field(self, ptype, fname, "code_mass")
             return
 
-        constants = nested_dict_get(
-            self.ds.parameters, ("Particle", ptype, "constants"), default=[]
-        )
+        pdict = nested_dict_get(self.ds.parameters, ("Particle"), default=None)
+        if pdict is None:
+            return
+
+        constants = nested_dict_get(pdict, (ptype, "constants"), default=())
         if not constants:
-            names = []
-        else:
-            if not isinstance(constants[0], tuple):
-                constants = (constants,)
-            names = [c[0] for c in constants]
-
-        if "mass" in names:
-            val = constants[names.index("mass")][2]
-            val = self.ds.quan(val, self.ds.mass_unit)
-            if self.ds.cosmological_simulation:
-                val = val / self.ds.domain_dimensions.prod()
-
-            def _pmass(field, data):
-                return val * data[ptype, "particle_ones"]
-
-            self.add_field(
-                (ptype, name),
-                function=_pmass,
-                units="code_mass",
-                sampling_type="particle",
+            return
+
+        if not isinstance(constants[0], tuple):
+            constants = (constants,)
+        names = [c[0] for c in constants]
```

Comment:
This looks broken to me, I only assume it's not because it's not new code.
Isn't it something that could be expressed with `more_itertools.always_iterable` ?

Reply:
It's not broken. The format of `constants` is meant to be a tuple consisting of multiple instances of (`name`, `type`, `value`) tuples. Unfortunately, when there is only one entry, the enclosing tuple is dropped and we end up with just the first entry. I don't think `always_iterable` will simplify that.

Reply:
I see. You're right that always_iterable will not help here. However, maybe some of your response could be useful as a comment here !

Reply:
Fair enough. Will do.